arcaflow_plugin_coremark_pro/coremark_pro_plugin.py

```diff
@@ -131,18 +133,20 @@ def certify_all(
         "CoreMark-PRO": {},
     }
 
+    # Construct the output object
     for line in ca_return[1].splitlines():
         line_list = line.split()
         try:
             line_name = line_list[0]
         except IndexError:
-            pass
-        if line_name in ca_results and len(line_list) > 1:
+            continue
+        if line_name in ca_results:
             ca_results[line_name] = {
                 "MultiCore": float(line_list[1]),
                 "SingleCore": float(line_list[2]),
                 "Scaling": float(line_list[3]),
             }
+            # Collect the per-benchmark iterations from the config files
```

Comment:
"from the config files"? This is reading a single *log* file, and apparently finding the "iterations" value specifically for "the" (or maybe "the last") line with "CoreMark-PRO" in the 3rd token. (And if you don't expect more than one, a `break` when you find it would clarify that.)

Is more explanation perhaps required here, or perhaps less??? 😆 It's not obvious to this reader what you expect to find in that log file, aside from the two loops where you read it, and frankly this comment makes your intent seem more mysterious than the uncommented code.

Reply:
I mixed up the write and read actions. We are writing to config files, but reading from the log file.